### ISSUE TITLE
File is not updated neither in repo nor in published version

### ISSUE THREAD

**Issue by sebastian-muthwill** [brettdorrans/write-version-to-file#5]
Probably I don´t understand the concept right. The documentation is also not really detailed and therefor not helpfull at all.

### Szenario 1
I set up the example within my ci pipeline once as it is. This will fetch the version from last tag correctly but the result is then
```
File Content: v0.10.1
Updating Version To: v0.10.3
Updated Content: v0.10.1
```
This has also no effect on the version file in the repo.

### Szenario 2
I integrated just the part:
```
- name: Update version
      id: version
      uses: brettdorrans/write-version-to-file@master
      with:
        filename: 'VERSION'
```
after the repo checkout in by build ci. In this case i would expect the version file is updated in the container and pushed afterwards directly to the production server. 
Within the build process I have the same result from the action:

```
File Content: v0.10.1
Updating Version To: v0.10.3
Updated Content: v0.10.1
```

But again the version file which is pushed is in both cases empty in production.  That means, the version has been deleted from the file during the build process? And why is the `Updated Content: v0.10.1` still the same as the `File Content: v0.10.1` after update?

## Suggestions

Provide a working example in documentation together with a explanation of what to expect as the output of this action.

**Comment by brettdorrans** [comment]
Hi @sebastian-muthwill - thanks for raising an issue. Hopefully I can clarify some things for you.

The script does not commit any changes to the repository, but only updates the VERSION file _within the CI container_, so build tooling can then access this information elsewhere. To give a concrete example, I previously used this to update a file with the latest version tag, which a CI build step then baked into a compiled version of the application (the file in question was directly imported into that project's build process, so the version appearing in the compiled version matched the latest git tag). Therefore no updating of the repository file is required as this is a transient update and only required for the next step of the build process.

In regards to your file not being updated with the latest git tag - can you link me to an example of this, if possible? I do not currently use this action myself.

**Comment by sebastian-muthwill** [comment]
HI @brettdorrans thanks for clarification. 
In the meantime I figured out, that following line 
`updatedContent=$(cat "$filename" | sed "s/\${VERSION}/$latestVersionTag/g")`
caused the empty file. I´m not shure where the `${VERSION}` in your case came from, but in my case this was empty, resulting in an empty file in the CI built container. 

I forked your repo and replaced mentioned line with
`updatedContent=$latestVersionTag`.
So basically I do not care what is in the version file, I just replace it with the new version derived from the tag. 
You can find it here: https://github.com/sebastian-muthwill/write-version-to-file/blob/36b0535cac22e92c411836a73d4b481e4ae43afb/entrypoint.sh#L17

I will update the readme in my fork and give credit to your work. 

From my side you can close the issue. 

Thanks for your support.
